util/sqlexec/utils_test.go

```diff
@@ -102,14 +96,17 @@ func (s *testUtilsSuite) TestEscapeBackslash(c *C) {
 			output: []byte("中文?"),
 		},
 	}
-	for _, t := range tests {
-		commentf := Commentf("%s", t.name)
-		c.Assert(escapeBytesBackslash(nil, t.input), DeepEquals, t.output, commentf)
-		c.Assert(escapeStringBackslash(nil, string(t.input)), DeepEquals, t.output, commentf)
+	for _, v := range tests {
+		t.Run(v.name, func(t *testing.T) {
+			t.Parallel()
+			require.Equal(t, v.output, escapeBytesBackslash(nil, v.input))
+			require.Equal(t, v.output, escapeStringBackslash(nil, string(v.input)))
+		})
 	}
 }
 
-func (s *testUtilsSuite) TestEscapeSQL(c *C) {
+func TestEscapeSQL(t *testing.T) {
+	t.Parallel()
```

Comment:
Maybe remove this line if you already parallel sub tests. **Not a necessary change request.**

If all sub tests are paralleled, this function will not wait for them, i.e. it will return once it submitted a goroutine for every sub test, which should be pretty fast.

It does not make much sense to parallel a short-live task: tasks will preempt resource with each other, which is not free on performance. **Lots of** parallel tasks can benefit from multiple cpu cores/threads, otherwise the cost of preemption is longer than the execution time of tasks.

So one more parallel task does not make much difference, but does get the resource preemption more intense. If the task firing sub tasks got paused due to preemption, the advantage from firing multiple parallel tasks may disappear.

However, for this case, every loop iteration is short, too, and there are not many loop iterations. So it does not make a big difference to parallel or not.

Reply:
get it, thank you for your detailed explain :heart: